File: csharp/raindrops/Raindrops.cs
using System.Globalization;

public static class Raindrops
{
    public static string Convert(int number)
    {
        var convertedNumber = string.Empty;

        if (number % 3 == 0)
        {
            convertedNumber += "Pling";
        }

        if (number % 5 == 0)
        {
            convertedNumber += "Plang";
        }

        if (number % 7 == 0)
        {
            convertedNumber += "Plong";
        }

        return convertedNumber == string.Empty ? number.ToString(CultureInfo.InvariantCulture) : convertedNumber;
    }
}

File: cpp/raindrops/raindrops.cpp
#include "raindrops.h"

#include <string>

namespace raindrops
{
    std::string convert(int n)
    {
        std::string result;

        if (n % 3 == 0)
            result.append("Pling");
        if (n % 5 == 0)
            result.append("Plang");
        if (n % 7 == 0)
            result.append("Plong");

        return result.empty() ? std::to_string(n) : result;
    }
} // namespace raindrops
